From 1fa6b660cb500a441defb168649d148c86347199 Mon Sep 17 00:00:00 2001
From: Haohui Mai <wheat9@apache.org>
Date: Tue, 29 Sep 2015 14:56:06 -0700
Subject: [PATCH 1090/2748] HDFS-9174. Fix findbugs warnings in
 FSOutputSummer.tracer and
 DirectoryScanner$ReportCompiler.currentThread.
 Contributed by Yi Liu.

(cherry picked from commit baa6ef020797ab2b562d48ce08585c4844b4755c)

Change-Id: I144b575df44071f3682d44bbc9dc1bf9c941962f
---
 .../java/org/apache/hadoop/fs/FSOutputSummer.java  |    3 ---
 .../hdfs/server/datanode/DirectoryScanner.java     |    8 --------
 2 files changed, 11 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FSOutputSummer.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FSOutputSummer.java
index 3a07cd6..6317bcc 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FSOutputSummer.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FSOutputSummer.java
@@ -22,7 +22,6 @@
 import org.apache.hadoop.classification.InterfaceStability;
 import org.apache.hadoop.util.DataChecksum;
 import org.apache.htrace.core.TraceScope;
-import org.apache.htrace.core.Tracer;
 
 import java.io.IOException;
 import java.io.OutputStream;
@@ -43,8 +42,6 @@
   private byte checksum[];
   // The number of valid bytes in the buffer.
   private int count;
-  // The HTrace tracer to use
-  private Tracer tracer;
   
   // We want this value to be a multiple of 3 because the native code checksums
   // 3 chunks simultaneously. The chosen value of 9 strikes a balance between
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DirectoryScanner.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DirectoryScanner.java
index d7ecc37..3fffd95 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DirectoryScanner.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DirectoryScanner.java
@@ -785,12 +785,6 @@ private static boolean isBlockMetaFile(String blockId, String metaFile) {
     private final StopWatch perfTimer = new StopWatch();
 
     /**
-     * The associated thread.  Used for testing purposes only.
-     */
-    @VisibleForTesting
-    Thread currentThread;
-
-    /**
      * Create a report compiler for the given volume on the given datanode.
      *
      * @param datanode the target datanode
@@ -809,8 +803,6 @@ public ReportCompiler(DataNode datanode, FsVolumeSpi volume) {
      */
     @Override
     public ScanInfoPerBlockPool call() throws IOException {
-      currentThread = Thread.currentThread();
-
       String[] bpList = volume.getBlockPoolList();
       ScanInfoPerBlockPool result = new ScanInfoPerBlockPool(bpList.length);
       for (String bpid : bpList) {
-- 
1.7.9.5

